crates/services/p2p/src/service.rs

```diff
@@ -425,6 +434,32 @@ impl SharedState {
     ) -> broadcast::Receiver<BlockHeightHeartbeatData> {
         self.block_height_broadcast.subscribe()
     }
+
+    pub fn report_peer<T: PeerReport>(
+        &self,
+        peer_id: FuelPeerId,
+        peer_report: T,
+        reporting_service: &'static str,
+    ) -> anyhow::Result<()> {
+        match Vec::from(peer_id).try_into() {
+            Ok(peer_id) => {
+                let peer_score = peer_report.get_score_from_report();
+
+                self.request_sender
+                    .try_send(TaskRequest::RespondWithPeerReport {
```

Comment:
What do you think about renaming `RespondWithPeerReport` -> `ApplicationReport`?

Reply:
I'd prefer keep `RespondWith` and as discussed, the report is about the peer,  `ApplicationReport` would suggest otherwise